docs/codeql/codeql-language-guides/data-extensions-to-model-java-dependencies.rst

```diff
@@ -0,0 +1,295 @@
+.. _data-extensions-to-model-dependencies:
```

Comment:
General comment: this article is covers several concepts:

* firstly, data extensions
* secondly, the format used to model sources, sinks and summaries in Java code

I _think_ the original title 'Customizing Library Models for Java" was slightly more appropriate to cover both of those concepts.